File: app-impl/src/main/java/be/insaneprogramming/cleanarch/interactor/EvictTenantFromBuildingImpl.java
package be.insaneprogramming.cleanarch.interactor;

import javax.inject.Named;
import javax.transaction.Transactional;

import be.insaneprogramming.cleanarch.boundary.EvictTenantFromBuilding;
import be.insaneprogramming.cleanarch.entity.Building;
import be.insaneprogramming.cleanarch.entitygateway.BuildingEntityGateway;
import be.insaneprogramming.cleanarch.requestmodel.EvictTenantFromBuildingRequest;

@Transactional
@Named
public class EvictTenantFromBuildingImpl implements EvictTenantFromBuilding {
	private final BuildingEntityGateway buildingEntityGateway;

	public EvictTenantFromBuildingImpl(BuildingEntityGateway buildingEntityGateway) {
		this.buildingEntityGateway = buildingEntityGateway;
	}

	@Override
	public void execute(EvictTenantFromBuildingRequest request) {
		Building building = buildingEntityGateway.findById(request.getBuildingId());
		building.evictTenant(request.getTenantId());
		buildingEntityGateway.save(building);
	}
}

File: app-impl/src/main/kotlin/be/insaneprogramming/cleanarch/interactor/EvictTenantFromBuildingImpl.kt
package be.insaneprogramming.cleanarch.interactor

import be.insaneprogramming.cleanarch.boundary.EvictTenantFromBuilding
import be.insaneprogramming.cleanarch.entity.BuildingId
import be.insaneprogramming.cleanarch.entitygateway.BuildingEntityGateway
import be.insaneprogramming.cleanarch.requestmodel.EvictTenantFromBuildingRequest

class EvictTenantFromBuildingImpl(private val buildingEntityGateway: BuildingEntityGateway) : EvictTenantFromBuilding {
    override fun execute(request: EvictTenantFromBuildingRequest) {
        val building = buildingEntityGateway.findById(BuildingId(request.buildingId))
        building.evictTenant(request.tenantId)
        buildingEntityGateway.save(building)
    }
}
